src/components/ha-form/ha-form-integer.ts

```diff
@@ -96,9 +96,9 @@ export class HaFormInteger extends LitElement implements HaFormElement {
   }
 
   private _valueChanged(ev: Event) {
-    const value = Number(
-      (ev.target as PaperInputElement | PaperSliderElement).value
-    );
+    const rawValue = (ev.target as PaperInputElement | PaperSliderElement)
+      .value;
+    const value = rawValue === "" ? "" : Number(rawValue);
```

Comment:
An integer should never be an empty string?

Reply:
this is for an optional integer. if empty, it doesn't fill that field in the data, but otherwise there is no way to distinguish an empty integer cell (ok if optional, disables submit if required) and a zero cell. today the behavior displays it as 0 and treats it as empty. this will end up with the required fields and the optional ones that were filled to be included in the data

Reply:
if should be undefined, not an empty string

Reply:
ok. changed. i thought that paper-input uses an empty string but in either case, both are treated by the code as empty.

Reply:
i just ran into a few issues with this change. 

the problem is that when paper-input for an integer is empty, it sends the value-changed event with "". Then the logic updates the data to delete this item, and now it gets value=undefined, which triggers another change. I am sure I can work around the logic and make it ok, but clearly paper-input is fine with getting value="" as an empty value for an integer, and even expects it (https://github.com/PolymerElements/paper-input/issues/633 shows that they accidentally changed the behavior and changed it back).
Please let me know if ok to use "", otherwise I will continue debugging :)